airflow-core/src/airflow/models/deadline.py

```diff
@@ -150,6 +160,36 @@ def prune_deadlines(cls, *, session: Session, conditions: dict[Column, Any]) ->
 
         return deleted_count
 
+    def handle_miss(self, session: Session):
+        """Handle a missed deadline by creating and starting a trigger to run the callback."""
+        callback_func = import_string(self.callback)
+        # TODO: Improve this check so that the callback doesn't need to be present on the scheduler
+        if not asyncio.iscoroutinefunction(callback_func):
+            # TODO: For sync callbacks, use executor instead of trigger
+            logger.error("Sync callbacks not supported yet: %s", self.callback)
```

Comment:
Could this be detected at parse time Instead of only when the callback is being run?